pkg/asset/installconfig/azure/dns.go

```diff
@@ -76,32 +78,50 @@ func (config DNSConfig) GetPrivateDNSZoneID(rgName string, zoneName string) stri
 func (config DNSConfig) GetDNSZone() (*Zone, error) {
 	//call azure api using the session to retrieve available base domain
 	zonesClient := newZonesClient(config.Session)
-	allZones, _ := zonesClient.GetAllPublicZones()
-	if len(allZones) == 0 {
+	publicZonesMap, err := zonesClient.GetAllPublicZones()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not retrieve base domains")
+	}
+
+	if len(publicZonesMap) == 0 {
 		return nil, errors.New("no public dns zone found in your subscription")
 	}
-	zoneNames := []string{}
-	for zoneName := range allZones {
-		zoneNames = append(zoneNames, zoneName)
+
+	publicZones := make([]string, 0, len(publicZonesMap))
+	for name := range publicZonesMap {
+		for _, id := range publicZonesMap[name] {
+			parts := strings.Split(id, "/")
+			rgName := parts[4]
+			publicZones = append(publicZones, fmt.Sprintf("%s (%s)", name, rgName))
```

Comment:
i think we can just keep the zone id intact here, wdyt? it keeps us from remaking the id again here https://github.com/openshift/installer/pull/2764/commits/b5b4736d951a8a004074c35ef13aebbc37024779#diff-b58e294d4fcb67a8f98ccb64bad839c1R117-R120

Reply:
but we can keep it as-is too, no major preference for me here.

Reply:
This same work needs to be done either here, or with the returned result. I don't have a preference either. I'll keep it here for now.